runtime/bundles/org.eclipse.core.contenttype/src/org/eclipse/core/internal/content/ContentType.java

```diff
@@ -377,15 +392,15 @@ boolean hasBuiltInAssociations() {
 		return builtInAssociations;
 	}
 
-	boolean hasFileSpec(IScopeContext context, String text, int typeMask) {
+	boolean hasFileSpec(IScopeContext context, String text, int typeMask, boolean caseStrict) {
```

Comment:
As we already have a typeMask, why not having a new flag controlling the behavior?